virt-density-udn.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2024 The Kube-burner Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ocp
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/kube-burner/kube-burner/pkg/workloads"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+// Returns virt-density workload
+func NewVirtUDNDensity(wh *workloads.WorkloadHelper) *cobra.Command {
+	var iterations int
+	var vmiRunningThreshold time.Duration
+	var metricsProfiles []string
+	var churnPercent, churnCycles int
+	var churn, l3 bool
+	var churnDelay, churnDuration time.Duration
+	var churnDeletionStrategy, jobPause, vmImage, bindingMethod string
+	var rc int
+	cmd := &cobra.Command{
+		Use:          "virt-udn-density",
+		Short:        "Runs virt-density workload",
+		SilenceUsage: true,
+		PreRun: func(cmd *cobra.Command, args []string) {
+			os.Setenv("JOB_PAUSE", jobPause)
+			os.Setenv("CHURN", fmt.Sprint(churn))
+			os.Setenv("CHURN_CYCLES", fmt.Sprintf("%v", churnCycles))
+			os.Setenv("CHURN_DURATION", fmt.Sprintf("%v", churnDuration))
+			os.Setenv("CHURN_DELAY", fmt.Sprintf("%v", churnDelay))
+			os.Setenv("CHURN_PERCENT", fmt.Sprint(churnPercent))
+			os.Setenv("CHURN_DELETION_STRATEGY", churnDeletionStrategy)
+			os.Setenv("JOB_ITERATIONS", fmt.Sprint(iterations))
+			os.Setenv("VMI_RUNNING_THRESHOLD", fmt.Sprintf("%v", vmiRunningThreshold))
+			os.Setenv("VM_IMAGE", vmImage)
+			os.Setenv("UDN_BINDING_METHOD", bindingMethod)
+		},
+		Run: func(cmd *cobra.Command, args []string) {
+			setMetrics(cmd, metricsProfiles)
+			if l3 {
+				log.Info("Layer 3 is enabled")
+				os.Setenv("ENABLE_LAYER_3", "true")
+				AddVirtMetadata(wh, vmImage, "layer3", bindingMethod)
+			} else {
+				log.Info("Layer 2 is enabled")
+				os.Setenv("ENABLE_LAYER_3", "false")
+				AddVirtMetadata(wh, vmImage, "layer2", bindingMethod)
+			}
+			rc = wh.Run(cmd.Name())
+		},
+		PostRun: func(cmd *cobra.Command, args []string) {
+			os.Exit(rc)
+		},
+	}
+	cmd.Flags().BoolVar(&l3, "layer3", true, "Layer3 UDN test")
+	cmd.Flags().BoolVar(&churn, "churn", false, "Enable churning")
+	cmd.Flags().IntVar(&churnCycles, "churn-cycles", 0, "Churn cycles to execute")
+	cmd.Flags().StringVar(&jobPause, "job-pause", "1ms", "Time to pause after finishing the job")
+	cmd.Flags().StringVar(&vmImage, "vm-image", "quay.io/openshift-cnv/qe-cnv-tests-fedora:40", "Vm Image to be deployed")
```

Comment:
to be consistent can we either change this to Raul's image or update the other job to this image?

Reply:
I will use Raul's image because it is accessible from a public dockerhub where thqe-cnv-tests-fedora is not.